src/DataProvider/OperationDataProvider.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataProvider;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Identifier\Normalizer\ChainIdentifierDenormalizer;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+
+/**
+ * @internal
```

Comment:
Has it's a mandatory dependency of `IriConverter`, and there is no related interface, it cannot be `@internal`.

Reply:
Also, please add some PHPDoc.